### PR TITLE
SCC-1075 | Update Main Address - Future Date Error Message

### PR DESCRIPTION
**Pull request by Neozxc**

### CODE REVIEW

File: test/pages/changeofaddress/FutureDateErrorPageSpec.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package pages.changeofaddress
+
+import models.backend.Address
+import pages.answers.changeaddress.FutureDateErrorPage
+import pages.behaviours.PageBehaviours
+
+
+class FutureDateErrorPageSpec extends PageBehaviours {
```
Comment (owen-opencast) [inline test/pages/changeofaddress/FutureDateErrorPageSpec.scala:24]: Redundant

**Review by owen-opencast — CHANGES_REQUESTED** [review]

File: app/pages/answers/changeaddress/FutureDateErrorPage.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package pages.answers.changeaddress
+
+import models.backend.Address
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object FutureDateErrorPage extends QuestionPage[Address] {
```
[the remark this answers is not in the document]
Comment (ccatosdevelopment) [reply app/pages/answers/changeaddress/FutureDateErrorPage.scala]: File deleted

File: conf/address.routes [hunk 3]
```diff
@@ -27,23 +27,25 @@ POST       /:addressType/change-enter-your-uk-address                 controller
...
 
-GET        /postal/remove-postal-address                                     controllers.changeaddress.RemovePostalAddressController.onPageLoad(mode: Mode = NormalMode)
-POST       /postal/remove-postal-address                                     controllers.changeaddress.RemovePostalAddressController.onSubmit(mode: Mode = NormalMode)
+GET        /postal/remove-postal-address                              controllers.changeaddress.RemovePostalAddressController.onPageLoad(mode: Mode = NormalMode)
+POST       /postal/remove-postal-address                              controllers.changeaddress.RemovePostalAddressController.onSubmit(mode: Mode = NormalMode)
 
 GET        /postal/change-remove-postal-address                               controllers.changeaddress.RemovePostalAddressController.onPageLoad(mode: Mode = CheckMode)
 POST       /postal/change-remove-postal-address                               controllers.changeaddress.RemovePostalAddressController.onSubmit(mode: Mode = CheckMode)
+
+GET        /:addressType/future-date-error                                    controllers.changeaddress.FutureDateErrorController.onPageLoad(mode: Mode = NormalMode, addressType: AddressType)
```
[the remark this answers is not in the document]
Comment (ccatosdevelopment) [reply conf/address.routes]: Route updated as suggested, plus related files renamed to match

File: test-utils/generators/PageGenerators.scala [hunk 4]
```diff
@@ -17,11 +17,14 @@
 package generators
 
 import org.scalacheck.Arbitrary
-import pages.answers.changeaddress.{IsNewAddressInUkPage, RemovePostalAddressPage, SelectYourAddressPage, WhatIsYourNewAddressNonUkPage, WhatIsYourNewAddressUkPage, WhenDidYouStartLivingHerePage}
+import pages.answers.changeaddress.{FutureDateErrorPage, IsNewAddressInUkPage, RemovePostalAddressPage, SelectYourAddressPage, WhatIsYourNewAddressNonUkPage, WhatIsYourNewAddressUkPage, WhenDidYouStartLivingHerePage}
 import pages.data.{CorrespondenceAddressPage, EmailPage, MaritalStatusPage, NamePage, PreferredNamePage, ResidentialAddressPage, TelephonePage, TitlePage}
 
 trait PageGenerators {
 
+  implicit lazy val arbitraryFutureDateErrorPage: Arbitrary[FutureDateErrorPage.type] =
```
[the remark this answers is not in the document]
Comment (ccatosdevelopment) [reply test-utils/generators/PageGenerators.scala]: Deleted

File: test-utils/generators/UserAnswersEntryGenerators.scala [hunk 5]
```diff
@@ -21,12 +21,20 @@ import models.frontend.RemovePostalAddress
...
 import org.scalacheck.Arbitrary.arbitrary
 import org.scalacheck.Gen.{chooseNum, oneOf}
-import pages.answers.changeaddress.{IsNewAddressInUkPage, RemovePostalAddressPage, SelectYourAddressPage, WhatIsYourNewAddressNonUkPage, WhatIsYourNewAddressUkPage, WhenDidYouStartLivingHerePage}
+import pages.answers.changeaddress.{FutureDateErrorPage, IsNewAddressInUkPage, RemovePostalAddressPage, SelectYourAddressPage, WhatIsYourNewAddressNonUkPage, WhatIsYourNewAddressUkPage, WhenDidYouStartLivingHerePage}
 import pages.data.{CorrespondenceAddressPage, EmailPage, MaritalStatusPage, NamePage, PreferredNamePage, ResidentialAddressPage, TelephonePage, TitlePage}
 import play.api.libs.json.{JsValue, Json}
 
 trait UserAnswersEntryGenerators extends PageGenerators with ModelGenerators {
 
+  implicit lazy val arbitraryFutureDateErrorUserAnswersEntry: Arbitrary[(FutureDateErrorPage.type, JsValue)] =
```
[the remark this answers is not in the document]
Comment (ccatosdevelopment) [reply test-utils/generators/UserAnswersEntryGenerators.scala]: Deleted

File: test-utils/generators/UserAnswersGenerator.scala [hunk 6]
```diff
@@ -21,14 +21,15 @@ import org.scalacheck.Arbitrary.arbitrary
...
-import pages.answers.changeaddress.{IsNewAddressInUkPage, RemovePostalAddressPage, SelectYourAddressPage, WhatIsYourNewAddressNonUkPage, WhatIsYourNewAddressUkPage, WhenDidYouStartLivingHerePage}
+import pages.answers.changeaddress.{FutureDateErrorPage, IsNewAddressInUkPage, RemovePostalAddressPage, SelectYourAddressPage, WhatIsYourNewAddressNonUkPage, WhatIsYourNewAddressUkPage, WhenDidYouStartLivingHerePage}
 import pages.data.{CorrespondenceAddressPage, EmailPage, MaritalStatusPage, NamePage, PreferredNamePage, ResidentialAddressPage, TelephonePage, TitlePage}
 import play.api.libs.json.{JsValue, Json}
 
 trait UserAnswersGenerator extends TryValues {
   self: Generators =>
 
   val generators: Seq[Gen[(QuestionPage[_], JsValue)]] =
+    arbitrary[(FutureDateErrorPage.type, JsValue)] ::
```
[the remark this answers is not in the document]
Comment (ccatosdevelopment) [reply test-utils/generators/UserAnswersGenerator.scala]: Deleted

File: test/pages/changeofaddress/FutureDateErrorPageSpec.scala (hunk 1 above)
Comment (ccatosdevelopment) [reply test/pages/changeofaddress/FutureDateErrorPageSpec.scala]: Deleted

File: test/controllers/changeaddress/YourAddressCannotBeUpdatedControllerSpec.scala [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import repositories.{AddressLockRepository, UserAnswersRepository, UserDataRepository}
+import services.ChocsService
+import views.html.changeaddress.YourAddressCannotBeUpdatedView
+
+class YourAddressCannotBeUpdatedControllerSpec extends SpecBase with MockitoSugar {
+
+  val mockService: ChocsService              = mock[ChocsService]
```
Comment (owen-opencast) [inline test/controllers/changeaddress/YourAddressCannotBeUpdatedControllerSpec.scala:33]: These mocks don't get passed in to the applicationBuilder - can they be removed?

**Review by owen-opencast — CHANGES_REQUESTED** [review]

**Review by owen-opencast — APPROVED** [review]

**Pull request merged**